packages/contracts-bedrock/src/safe/LivenessModule2.sol

```diff
@@ -0,0 +1,295 @@
+        uint256 startTime = challengeStartTime[_safe];
+        if (startTime == 0) {
+            return 0;
+        }
+        ModuleConfig storage config = safeConfigs[_safe];
+        return startTime + config.livenessResponsePeriod;
+    }
+
+    /// @notice Configures the module for a Safe that has already enabled it
+    /// @dev MUST only be callable by a Safe that has enabled this module
+    /// @param _config The configuration parameters for the module
+    function configure(ModuleConfig memory _config) external {
+        // Validate configuration parameters to ensure module can function properly.
+        // livenessResponsePeriod must be > 0 to allow time for Safe owners to respond.
+        if (_config.livenessResponsePeriod == 0) {
+            revert LivenessModule2_InvalidResponsePeriod();
+        }
+        // fallbackOwner must not be zero address to have a valid ownership recipient.
+        if (_config.fallbackOwner == address(0)) {
+            revert LivenessModule2_InvalidFallbackOwner();
+        }
+
+        // Check that this module is enabled on the calling Safe.
+        // This prevents Safes from configuring themselves for a module they haven't enabled,
+        // which would create orphaned configuration data. The Safe must first call
+        // enableModule() before configure() to ensure proper two-step initialization.
+        Safe safe = Safe(payable(msg.sender));
+        if (!safe.isModuleEnabled(address(this))) {
+            revert LivenessModule2_ModuleNotEnabled();
+        }
+
+        // Store the configuration for this safe
+        safeConfigs[msg.sender] = _config;
+
+        // Clear any existing challenge when configuring/re-configuring.
+        // This is necessary because changing the configuration (especially livenessResponsePeriod)
+        // would invalidate any ongoing challenge timing, creating inconsistent state.
+        // For example, if a challenge was started with a 7-day period and we reconfigure to
+        // 1 day, the challenge timing becomes ambiguous. Canceling ensures clean state.
+        _cancelChallenge(msg.sender);
+
+        emit ModuleEnabled(msg.sender, _config.livenessResponsePeriod, _config.fallbackOwner);
+    }
+
+    /// @notice Clears the module configuration for a Safe
+    /// @dev MUST only be executable by a Safe that has DISABLED this module first
+    /// @dev MUST erase existing liveness_response_period and fallback_owner data
+    /// @dev Note: Clearing the configuration also cancels any ongoing challenges
+    function clear() external {
+        // Check if the calling safe has configuration set
+        _assertModuleConfigured(msg.sender);
+        // Check that this module is NOT enabled on the calling Safe
+        // This prevents clearing configuration while module is still enabled
+        Safe safe = Safe(payable(msg.sender));
+        if (safe.isModuleEnabled(address(this))) {
```

Comment:
Might as well have `_assertModuleNotEnabled` I guess

Reply:
Fixed in e03a5d7dd728851d60cf929827f0cbec5f1803ee